src/dialects/abstract/query-generator.js

```diff
@@ -2224,17 +2224,22 @@ export class AbstractQueryGenerator extends AbstractQueryGeneratorTypeScript {
     }
   }
 
-  _getSelectOptimizerHints(_options) {
-    return 'SELECT';
+  _getBeforeSelectFragment(_options) {
+    return '';
+  }
+
+  _getAfterSelectFragment(_options) {
+    return '';
   }
 
   selectFromTableFragment(options, model, attributes, tables, mainTableAs) {
     this._throwOnEmptyAttributes(attributes, { modelName: model && model.name, as: mainTableAs });
 
     this._validateSelectOptions(options);
 
-    let fragment = this._getSelectOptimizerHints(options);
-
+    let fragment = this._getBeforeSelectFragment(options);
```

Comment:
Wait but actually we don't need this. That can already be done by overriding `selectFromTableFragment` to return something extra at the beginning of the string

Only `_getAfterSelectFragment` is necessary

Reply:
I don't understand about you said only `_getAfterSelectFragment` is necessary. I think if we will overriding `selectFromTableFragment` to return something extra at the beginning of the string, it can be return a something extra at the end of the string too.

In that context, Is `_getAfterSelectFragment` unnecessary too?

Reply:
In the code you had, `_getAfterSelectFragment` wasn't added to the end of the string but inside of it

To be sure there isn't a misunderstanding: I'm talking about `_getBeforeSelectAttributesFragment`, which was called `_getAfterSelectFragment` before. A `_getAfterSelectFragment` that's added at t he end of the string would be unnecessary too

Reply:
Ok, now I understand why `_getAfterSelectFragment(same with _getBeforeSelectAttributesFragment)` is necessary.

Then after we delete unnecessary `_getBeforeSelectFragment` function, how we support optimizer hint in MariaDB? By overriding `selectFromTableFragment`?

I think overriding `selectFromTableFragment` is take more cost than overriding `_getBeforeSelectFragment`. Because if we override `selectFromTableFragment`, we need to implement entire `selectFromTableFragment` logic.(I think `selectFromTableFragment` have more responsibility than `_getBeforeSelectFragment`)

But if we use `_getBeforeSelectFragment` method, we can support that by implement(override) `_getBeforeSelectFragment` only in MariaDB query generator.

Reply:
Ok, now I understand why `_getAfterSelectFragment(same with _getBeforeSelectAttributesFragment)` is necessary.

Then after we delete unnecessary `_getBeforeSelectFragment` function, how we support optimizer hint in MariaDB? By overriding `selectFromTableFragment`?

I think overriding `selectFromTableFragment` is take more cost than overriding `_getBeforeSelectFragment`. Because if we override `selectFromTableFragment`, we need to implement entire `selectFromTableFragment` logic.(I think `selectFromTableFragment` have more responsibility than `_getBeforeSelectFragment`)

But if we use `_getBeforeSelectFragment` method, we can support that by implement(override) `_getBeforeSelectFragment` only in MariaDB query generator.

@ephys Thanks for advise.

Reply:
> I think overriding selectFromTableFragment is take more cost than overriding _getBeforeSelectFragment. Because if we override selectFromTableFragment, we need to implement entire selectFromTableFragment logic.

not at all, just need to do this:

```ts
class MariaDbQueryGenerator extends MySqlQueryGenerator {
  selectFromTableFragment(options, model, attributes, tables, mainTableAs) {
    return 'SET ...' + super.selectFromTableFragment(options, model, attributes, tables, mainTableAs);
  }
}
```

Reply:
I was dumb.